notes-0319-inductive-types.tex

```diff
@@ -0,0 +1,228 @@
+\documentclass[11pt]{article}
+\usepackage{hdtt2020}
+\usepackage{newpxmath}
+\usepackage{newpxtext}
+
+\vspace{1mm}
+
+\newcommand{\N}{\mathbb{N}}
+\newcommand{\2}{\textbf{2}}
+\newcommand{\Sp}{\mathbb{S}^1}
+
+\title{CSCI 8980 Higher-Dimensional Type Theory\\ Lecture Notes}
+\author{Calvin Roth and Devon Tuma}
+\date{March 27, 2020}
+
+\begin{document}
+
+\maketitle
+
+\section{Wrap Up on Universes}
+
+\begin{theorem}
+The universe type $U$ is not a set. 
+\end{theorem}
+\begin{proof}
+Consider the negation function on $\2$, which is an equivalence between $\2$ and $\2$.
+This gives rise to some path \textit{neg} by univalence. 
+Then if this path were identified with \textit{refl}, we would have point-wise identifications between the identity and negation functions, and in particular an identification between their images at $0$, namely an element of type $Id_2(0,1)$.
+But in homework $3$ we showed that from this we can derive an element of $\bot$, so we have a contradiction.
+So $U$ can't be a set.
+\end{proof}
+
+\section{Inductive Type}
+We define an inductive type to be a type that is freely generated by a finite set of constructors, where intuitively we think of ``freely generated'' to mean that the elements of the type are all the objects that can be obtained by applying and composing the constructors. It also means that we don't impose any more relations on elements than we are absolutely required to. If you are familiar with freely generated groups from algebra then this should feel very similar.
+
+\begin{example}
+  $\2$ is generated by $0 : \2$ and $1 : \2$. Since neither constructor is a function type, there are no operations we can perform on these constructors. Therefore the type freely generated by these constructors has only two elements, as we would expect.
+\end{example}
+
+\begin{example}
+  $A + B$ is generated by $inl : A \to A + B$ and $inr : B \to A + B$. In this case the constructors are functions, so we get some more interesting elements of the inductive type. The type freely generated by these constructors contains elements like $inl\ a$ for elements $a : A$. 
```

Comment:
This is minor, but you could consider these two macros:
https://github.com/favonia/hdtt2020-notes/blob/5a2e3cb075d3f8fa67cc8aba2ca3003df2b2c20c/hdtt2020.sty#L110-L111